apps/api/src/app/workflows-v2/usecases/get-workflow/get-workflow.usecase.ts

```diff
@@ -0,0 +1,97 @@
+import { WorkflowTemplateGetMapper } from '../../mappers/workflow-template-get-mapper';
+
+@Injectable()
+export class GetWorkflowUseCase {
+  constructor(
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private environmentRepository: EnvironmentRepository,
+    private controlValuesRepository: ControlValuesRepository
+  ) {}
+  async execute(command: GetWorkflowCommand): Promise<WorkflowResponseDto> {
+    await this.validateEnvironment(command);
+    const notificationTemplateEntity = await this.notificationTemplateRepository.findByIdQuery({
+      id: command._workflowId,
+      environmentId: command.user.environmentId,
+    });
+
+    if (notificationTemplateEntity === null || notificationTemplateEntity === undefined) {
+      throw new BadRequestException(`Workflow not found with id: ${command._workflowId}`);
+    }
+    const stepIdToControlValuesMap = this.getControlsValuesMap(notificationTemplateEntity.steps, command);
+    const workflow = WorkflowTemplateGetMapper.toResponseWorkflowDto(notificationTemplateEntity);
+    //
+    return { ...workflow, steps: stepDtos };
+  }
+
+  private isContainValues(
+    stepControlsVariables: ControlValuesEntity | null
+  ): stepControlsVariables is ControlValuesEntity & { controls: Record<string, unknown> } {
+    if (!stepControlsVariables?.controls) {
+      return false;
+    }
+
+    return Object.keys(stepControlsVariables?.controls).length > 0;
+  }
+
+  private async validateEnvironment(command: GetWorkflowCommand) {
+    const environment = await this.environmentRepository.findOne({ _id: command.user.environmentId });
+
+    if (!environment) {
+      throw new BadRequestException('Environment not found');
+    }
+  }
+
+  private async getControlsValuesMap(
+    steps: NotificationStepEntity[],
+    command: GetWorkflowCommand
+  ): Promise<{ [key: string]: ControlValuesEntity }> {
+    const acc: { [key: string]: ControlValuesEntity } = {};
+
+    for (const step of steps) {
+      let controlValuesEntity = await this.buildControlVariablesforStep(step, command);
+      if (controlValuesEntity) {
+        acc[step._templateId] = controlValuesEntity;
+      }
+    }
+
+    return acc; // Return the accumulated results
+  }
+  private async buildControlVariablesforStep(
+    step: NotificationStepEntity,
+    command: GetWorkflowCommand
+  ): Promise<ControlValuesEntity | null> {
+    const defaultSchema = GetStepControlSchemaUsecase();
+    const defaultControlValues = defaults(defaultSchema);
+
+    return await this.controlValuesRepository.findOne({
+      _environmentId: command.user.environmentId,
+      _organizationId: command.user.organizationId,
+      _workflowId: command._workflowId,
+      stepId: step._templateId,
+      level: ControlVariablesLevelEnum.STEP_CONTROLS,
+    });
+  }
+}
+
+const GetStepControlSchemaUsecase = () => {
+  // tmp solution, will be replaced with the dynamic control schema once PR 6482 is merged
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
☑ **chore:** Please use the `TODO: ` directive to ensure such temporary solutions can easily be identified by search.‏